sklearn/utils/_param_validation.py

```diff
@@ -443,3 +444,44 @@ def generate_invalid_param_val(constraint):
             return interval.right + 1
     else:
         raise NotImplementedError
+
+
+def generate_valid_param(constraint):
```

Comment:
I assume that this function will change in the future because they are some cases that we don't think about right away and will come at the moment of validating the parameters of functions?

Reply:
True, the most likely to happen is the need of a new Constraint class